yarn-project/stdlib/src/block/l2_block_downloader/l2_block_stream.ts

```diff
@@ -93,17 +95,25 @@ export class L2BlockStream {
         this.hasStarted = true;
       }
 
+      let nextBlockNumber = latestBlockNumber + 1;
```

Comment:
I changed the while loop to work with `nextBlockNumber` instead of `latestBlockNumber` as that aligns much better with the notion of deciding where to start syncing from.